modules/installation-aws-regions-with-no-ami.adoc

```diff
@@ -1,17 +1,31 @@
 // Module included in the following assemblies:
 //
+// * installing/installing_aws/installing-aws-china.adoc
 // * installing/installing_aws/installing-aws-government-region.adoc
 // * installing/installing_aws/installing-aws-user-infra.adoc
 
+ifeval::["{context}" == "installing-aws-china-region"]
+:aws-china:
+endif::[]
+ifeval::["{context}" == "installing-aws-government-region"]
+:aws-gov:
+endif::[]
+
 [id="installation-aws-regions-with-no-ami_{context}"]
+ifndef::aws-china,aws-gov[]
 = AWS regions without a published {op-system} AMI
+endif::aws-china,aws-gov[]
+
+ifdef::aws-china,aws-gov[]
+= Installation requirments
```

Comment:
In this section, we are talking about two things, one is `rhcos` and the other is `install-config.yaml`.

It looks like the content is a bit overlapping:
 * section `Installation requirments` and `AWS government and secret regions`.
 * section `Installation requirments` and `Manually creating the installation configuration file`.

Section `AWS government and secret regions` and `Manually creating the installation configuration file` had been published in [4.8 doc](https://docs.openshift.com/container-platform/4.8/installing/installing_aws/installing-aws-government-region.html#installation-initializing-manual_installing-aws-government-region), to keep it consistent,  I have some suggestions:
1. add `AWS China regions  section` (according to `AWS government and secret regions`)
2. remove `Installation requirments` section.

WDYT?

Reply:
**Summary**:
I propose leaving the `Installation requirements` section and backing porting the changes to the gov and secret regions topic in 4.6, 4.7, and 4.8. I can take the item to create the BZ for that work.

**Reasoning**:
In the 4.8 version of the gov and secret region doc, the `AWS government and secret regions` section [1] mentions the requirement, but IMO, the req gets lost at the bottom of the paragraph. Given that the reqs are a fairly significant departure from what existing AWS customers are used to when deploying to a public region, I believe the content warrants its own section.  Note that I have already updated the 4.9 gov and secret regions doc to be consistent [2].

Yes - the `Installation requirements` and `Manually creating the installation configuration file` do overlap, but I do believe that repeating the content is warranted. The sections are not repeating paragraph upon paragraph of doc, but rather compliant each other in reminding users of the req. 

[1] https://docs.openshift.com/container-platform/4.8/installing/installing_aws/installing-aws-government-region.html#installation-aws-about-government-region_installing-aws-government-region
[2] https://deploy-preview-35550--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-government-region.html#installation-aws-regions-with-no-ami_installing-aws-government-region

Reply:
Mike, get it, thanks for detailed explanation.

Another concern is `You cannot use the OpenShift Container Platform installation program to create the installation configuration file. `, there are two option for preparing `install-config.yaml`

* option A:
1. create `install-config.yaml` manually from a empty file (like section `Manually creating the installation configuration file`)

* option B:
1. create `install-config.yaml` by installer, (select any region)
2. edit `install-config.yaml`, adding rhcos AMI and update region (since there is no China regions in the prompt )

so I think the description `You cannot use the OpenShift Container Platform installation program to create the installation configuration file. ` is not precise, could we delete this sentence or change another way to describe?

Reply:
@codyhoag and I had this discussion, as well. Originally, I was going to leave an existing note in `Manually creating the installation configuration file` [1] that states you can use the installer to create the configuration file and then modify the region after the fact. When documenting the gov/secret region feature, Cody had the same question but was told that using the installer was not possible because the exported SSH keys would be for GovCloud and a public region could not be selected. We extrapolated that comment[2] to also apply to the China regions.

In your testing, were you able to use the installer to create the installation configuration and subsequently update the region? If so, I will remove the note and update any other applicable content accordingly.

[1] https://docs.openshift.com/container-platform/4.8/installing/installing_aws/installing-aws-government-region.html#installation-initializing-manual_installing-aws-government-region
[2] https://github.com/openshift/openshift-docs/pull/25652#issuecomment-706308441

Reply:
@yunjiang29 thanks for the background on this. Engineering has been consistent with us promoting the manual creation option for the install-config. This actually came up recently again for ASH, where we were told again to opt for the manual way.

If it is **possible** to successfully install with an auto-generated install-config, I don't have a problem toning down the language so we don't say it's impossible. If a user wants to go off the given path and do it differently, that's fine; we just don't want to recommend it if it's not best practice :slightly_smiling_face:

Reply:
@mjpytlak @codyhoag thanks for your detailed explanation, I agree with you that provide the best practice for users, `option B` is simple and straight forward, but `option A` is a little _tricky_, it may get some of users confused. 

For this official documents, I prefer `option B`.


> If it is possible to successfully install with an auto-generated install-config, I don't have a problem toning down the language so we don't say it's impossible. If a user wants to go off the given path and do it differently, that's fine; we just don't want to recommend it if it's not best practice

I understand, in QE's CI pipeline, we always prepare the `install-config.yaml` from sketch (easy to implement and maintain in program).
I could do some tests based on this scenario, but even it get passed, it looks like not a best practice.

Reply:
@yunjiang29 what are your thoughts on proceeding with `option B` for GA and following up later for possibly promoting `option A`? The reason I say this is because there are several installer instructions outside of AWS China regions that promote the manual method (AWS GovCloud and Secret, Azure MAG, etc.). I would like to get a consensus from Engineering and QE that all of them should follow the installer auto-generated method, and wholesale update the relevant docs so they're all consistent. With docs freeze quickly approaching, I'm not sure if doing that now (with the required testing) is wise. What do you think?

Reply:
@mjpytlak  @codyhoag sorry for the confused info, let me clarify my point here, `option A` (current description in the document) looks good to me for GA, I think this is the best practice for all platform.